strapi/strapi
# feat(history): display new and unknown fields

### What does it do?
Putting this in review as a draft now since I will be off tomorrow. There is some outstanding work and I made one questionable decision that is [up for discussion](https://github.com/strapi/strapi/pull/20039#discussion_r1552261273).

- Display deleted attributes in an “unknown fields” section
- Display new attributes as a “New field” warning alert instead of an input

### Why is it needed?
- To communicate to the user the differences between the version’s content-type schema and current document’s content-type schema
- To make it clear what will an will not be restored

### How to test it?
- Go to kitchensink and create at least one version
- Now go to the schema.json and delete a bunch of fields
- Create another history version
- Go to the history page and go to your previous version, you should have an unknown fields sections with all the attributes you deleted
- Go to the schema.json and add a new attribute
- Create another history version
- Go back to the history page, the new attribute should display on the current version as normal, on the previous versions it should display as a warning alert
- Go to the schema.json and rename an attribute
- Create a new history version
- Go back to the history page, the renamed Attribute should display on the current version as normal, on previous versions it should display the renamed attribute as a “new field”, it should display the attribute with its old name in the “unknown fields” section

It should support all types including components and dynamic zones however currently for these types the value of those inputs is not displayed and they are always displayed in their empty state input. This is not caused by this PR and should be addressed in another PR.


TODO
- [x] Fix type errors
